### PR TITLE
Make ConflictSet abstract

### PR DESCRIPTION
**Pull request by edsko**
This commit is a pure refactoring, no semantic changes. Submitting as separate PR at @kosmikus  request.

This moves `ConflictSet` and `Var` into their own modules.

### CODE REVIEW

File: cabal-install/tests/UnitTests/Distribution/Client/Dependency/Modular/Solver.hs [hunk 1]
```diff
@@ -507,6 +507,7 @@ db15 = [
   , Right $ exAv   "E" 1            [ExFix "C" 2]
   ]
 
+
```
Comment (ezyang) [inline cabal-install/tests/UnitTests/Distribution/Client/Dependency/Modular/Solver.hs]: kill me

**Comment by ezyang** [comment]
LGTM.

**Comment by kosmikus** [comment]
This is great, thanks. Please go ahead and merge.

**Comment by edsko** [comment]
CI test failure seems to be transient?

**Comment by 23Skidoo** [comment]
Restarted the AppVeyor build.

**Comment by 23Skidoo** [comment]
Build bots are green now.

**Comment by edsko** [comment]
Merging since there are no objections.

**Pull request merged**